src/acquisition/covid_hosp/common/database.py

```diff
@@ -12,50 +11,40 @@
 # first party
 import delphi.operations.secrets as secrets
 from delphi.epidata.common.logger import get_structured_logger
-
-Columndef = namedtuple("Columndef", "csv_name sql_name dtype")
+from delphi.epidata.common.covid_hosp.covid_hosp_schema_io import CovidHospSomething
 
 class Database:
 
-  def __init__(self,
-               connection,
-               table_name=None,
-               hhs_dataset_id=None,
-               columns_and_types=None,
-               key_columns=None):
-    """Create a new Database object.
+  DATASET_NAME = None
 
-    Parameters
-    ----------
-    connection
-      An open connection to a database.
-    table_name : str
-      The name of the table which holds the dataset.
-    hhs_dataset_id : str
-      The 9-character healthdata.gov identifier for this dataset.
-    columns_and_types : tuple[str, str, Callable]
-      List of 3-tuples of (CSV header name, SQL column name, data type) for
-      all the columns in the CSV file.
+  def __init__(self):
     """
-
-    self.connection = connection
-    self.table_name = table_name
-    self.hhs_dataset_id = hhs_dataset_id
-    self.publication_col_name = "issue" if table_name == 'covid_hosp_state_timeseries' or table_name == "covid_hosp_state_daily" else \
-      'publication_date'
-    self.columns_and_types = {
-      c.csv_name: c
-      for c in (columns_and_types if columns_and_types is not None else [])
-    }
-    self.key_columns = key_columns if key_columns is not None else []
+    Create a new Database object.
+    """
+    # Make sure connection is reset - set this in connect()
+    self.connection = None
+
+    if self.DATASET_NAME is None:
+      raise NameError('no dataset given!') # Must be defined by subclasses
+
+    chs = CovidHospSomething()
+    # The name of the table which holds the dataset.
+    self.table_name = chs.get_ds_table_name(self.DATASET_NAME)
+    # The 9-character healthdata.gov identifier for this dataset.
+    self.hhs_dataset_id = chs.get_ds_dataset_id(self.DATASET_NAME)
+    self.metadata_id = chs.get_ds_metadata_id(self.DATASET_NAME)
+    self.issue_column = chs.get_ds_issue_column(self.DATASET_NAME)
+    # List of 3-tuples of (CSV header name, SQL column name, data type) for all the columns in the CSV file.
+    self.columns_and_types = {c.csv_name: c for c in chs.get_ds_ordered_csv_cols(self.DATASET_NAME)}
+    self.key_columns = chs.get_ds_key_cols(self.DATASET_NAME)
+    self.aggregate_key_columns = chs.get_ds_aggregate_key_cols(self.DATASET_NAME)
 
   @classmethod
   def logger(database_class):
     return get_structured_logger(f"{database_class.__module__}")
 
-  @classmethod
   @contextmanager
-  def connect(database_class, mysql_connector_impl=mysql.connector):
+  def connect(self, mysql_connector_impl=mysql.connector):
```

Comment:
I think this matches Python's naming convention for context managers - compare:
```
with f = open("/etc/filename"):
   # do stuff with a file, then the context manager closes it
```
If you'd like to rename the method, I'd do `create_connection` (like `socket.create_connection`)

Reply:
theres a difference in that `open()` and `socket.create_connection()` can be used outside of a context manager's `with` syntax:
```
with open('file') as f:
  f.blah()
```
vs
```
f = open('file')
f.blah()
f.close()
```
we dont have that option here, this `connect()` method must be used as a context manager.

but then again, its just a name...  ¯\\\_(ツ)_/¯

Reply:
I don't mind renaming this one either :) would prefer a verb rather than noun as a name, though; `wrapped_connection` feels like a class property (but `wrap_connection` could work fine!)